docs/reference/mapping/types/parent-join.asciidoc

```diff
@@ -0,0 +1,399 @@
+// TEST[continued]
+
+<1> Querying the `parent id` field (also see the <<query-dsl-has-parent-query,`has_parent` query>> and the <<query-dsl-has-child-query,`has_child` query>>)
+<2> Aggregating on the `parent id` field (also see the <<search-aggregations-bucket-children-aggregation,`children`>> aggregation)
+<3> Accessing the parent id` field in scripts
+
+
+==== Global ordinals
+
+The `join` field uses <<eager-global-ordinals,global ordinals>> to speed up joins.
+Global ordinals need to be rebuilt after any change to a shard. The more
+parent id values are stored in a shard, the longer it takes to rebuild the
+global ordinals for the `join` field.
+
+Global ordinals, by default, are built eagerly: if the index has changed,
+global ordinals for the `join` field will be rebuilt as part of the refresh.
+This can add significant time to the refresh. However most of the times this is the
+right trade-off, otherwise global ordinals are rebuilt when the first parent-join
+query or aggregation is used. This can introduce a significant latency spike for
+your users and usually this is worse as multiple global ordinals for the `join`
+field may be attempt rebuilt within a single refresh interval when many writes
+are occurring.
+
+When the `join` field is used infrequently and writes occur frequently it may
+make sense to disable eager loading:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "doc": {
+      "properties": {
+        "my_join_field": {
+          "type": "join",
+          "my_parent": "my_child",
+          "eager_global_ordinals": false
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+The amount of heap used by global ordinals can be checked per parent relation
+as follows:
+
+[source,sh]
+--------------------------------------------------
+# Per-index
+GET _stats/fielddata?human&fields=my_join_field#my_parent
+
+# Per-node per-index
+GET _nodes/stats/indices/fielddata?human&fields=my_join_field#my_parent
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+==== Multiple levels of parent join
+
+It is also possible to define multiple children for a single parent:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "doc": {
+      "properties": {
+        "my_join_field": {
+          "type": "join",
+          "my_parent": ["my_child", "another_child"]  <1>
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> `my_parent` is parent of `my_child`.
+
+... and multiple levels of parent/child:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "doc": {
+      "properties": {
+        "my_join_field": {
+          "type": "join",
+          "my_parent": ["my_child", "another_child"],  <1>
+          "another_child": "grand_child" <2>
```

Comment:
It isn't clear to me from reading the docs what this does. Does it make something like:

```
     my_parent
     /      |
    /       |
my_child  another_child
            |
            |
       grand_child
```
?